media/system/js/fields/permissions.js

```diff
@@ -7,132 +7,140 @@
+		}
+		else if (extension != false && view != false){
+			asset = extension + '.' + view + '.' + Joomla.getUrlParam('id');
+			title = document.getElementById('jform_title').value;
+		}
+		else if (extension == false && view != false){
+			asset = option + '.' + view + '.' + Joomla.getUrlParam('id');
+			title = document.getElementById('jform_title').value;
 		}
 
-		// Render messages, if any. There are only message in case of errors.
-		if (typeof response.messages == 'object' && response.messages !== null)
-		{
-			Joomla.renderMessages(response.messages);
+		var id = this.id.replace('jform_rules_', '');
+		var lastUnderscoreIndex = id.lastIndexOf('_');
+
+		var permission_data = "data=" +
+			"&comp=" + asset +
+			"&action=" + id.substring(0, lastUnderscoreIndex) +
+			"&rule=" + id.substring(lastUnderscoreIndex + 1) +
+			"&value=" + value +
+			"&title=" + title;
+
+		// Remove js messages, if they exist.
+		Joomla.removeMessages();
+
+		// doing ajax request
+		Joomla.request({
+			method: 'POST',
+			url: document.getElementById('permissions-sliders').getAttribute('data-ajaxuri'),
+			data: permission_data,
+			perform: true,
+			headers: {'Content-Type': 'application/x-www-form-urlencoded'},
+
+			onError: function (xhr) {
+				// Remove the spinning icon.
+				icon.removeAttribute('style');
+
+				Joomla.renderMessages(Joomla.ajaxErrorsMessages(xhr));
+
+				window.scrollTo(0, 0);
 
-			if (response.data && response.data.result == true)
-			{
-				icon.setAttribute('class', 'fa fa-check');
-			}
-			else
-			{
 				icon.setAttribute('class', 'fa fa-times');
+			},
+			onSuccess: function (response, xhr) {
+				// Remove the spinning icon.
+				icon.removeAttribute('style');
+				response = JSON.parse(response);
+
+				if (response.data) {
+					// Check if everything is OK
+					if (response.data.result === true) {
+						icon.setAttribute('class', 'fa fa-check');
+						var next = event.target.parentElement.nextElementSibling;
+						while (next.nodeName.toLowerCase() !== "td") {
+							next = next.nextElementSibling;
+						}
+
+						var span = next.querySelector('span');
+						span.removeAttribute('class');
+						span.innerHTML = '';
+						span.setAttribute('class', response.data.class);
+						span.innerHTML = response.data.text;
+
+					}
+				}
+
+				// Render messages, if any. There are only message in case of errors.
+				if (typeof response.messages == 'object' && response.messages !== null) {
+					Joomla.renderMessages(response.messages);
+
+					if (response.data && response.data.result == true) {
+						icon.setAttribute('class', 'fa fa-check');
+					}
+					else {
+						icon.setAttribute('class', 'fa fa-times');
+					}
+
+					window.scrollTo(0, 0);
+				}
 			}
+		});
+	}
+})(Joomla);
 
-			window.scrollTo(0, 0);
-		}
-	});
-}
 
 /**
  * Function to get parameters out of the url
  */
-function getUrlParam(variable) {
+Joomla.getUrlParam = function(variable) {
```

Comment:
can you move this to the IIFE function above?

Reply:
Any specific reason for a move this function? or that is the IIFE function standard?

Reply:
Just IIFE for all the code in a file ;)